hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleAccountState.java

```diff
@@ -177,7 +177,9 @@ public void deserialize(SerializableDataInputStream in, int version) throws IOEx
 		if (version >= RELEASE_0230_VERSION) {
 			cryptoAllowances = deserializeCryptoAllowances(in);
 			fungibleTokenAllowances = deserializeFungibleTokenAllowances(in);
-			nftAllowances = deserializeNftAllowances(in);
```

Comment:
⚠️ ⚠️ Although this will work when deserializing a 0.23.x or 0.24.x state that had 0 NFT allowances, it will ofc **not** work for NFT allowances serialized using the previous `Map`-based approach [here](https://github.com/hashgraph/hedera-services/blob/ae3db581654ac5a272166e0864bf093776e41a0c/hedera-node/src/main/java/com/hedera/services/utils/SerializationUtils.java#L76).

Luckily, only previewnet can have any accounts with a non-zero number of NFT allowances in state.

We should check with Atul that it's ok for us to just deserialize those allowances, then throw them away:
```
...
	if (version >= RELEASE_0230_VERSION) {
		cryptoAllowances = deserializeCryptoAllowances(in);
		fungibleTokenAllowances = deserializeFungibleTokenAllowances(in);
                if (version < RELEASE_0250_VERSION) {
                       // Just forget < 0.25.x previewnet state (this will be empty for all other production networks)
                       deserializeLegacyNftAllowances(in); 
                } else {
                       approveForAllNfts = deserializeApproveForAllNftsAllowances(in);
                }
	}
...
```

Reply:
Yup, this was one of the issues me and @Neeharika-Sompalli discussed. forgot to place a marker here for clarification.